src/controllers/requestController.js

```diff
@@ -82,5 +84,67 @@ class Requests {
       return Response.errorResponse(res, 500, 'Something went wrong', error);
     }
   }
+
+  /**
+   * @param {object} req request
+   * @param {object} res response
+   * @return {function} requests
+   */
+  async rejectRequest(req, res) {
+    try {
+      const { reason } = req.body;
+      const manager = req.user;
+      const { requestId } = req.params;
+      const request = await requestService.getRequest({ id: requestId });
+      if (!request) {
+        return Response.errorResponse(res, 404, 'request not found', 'Not found');
+      }
+      const requesterId = request.user;
+      if (request.status === 'Approved') {
+        return Response.errorResponse(res, 409, 'Request already approved', 'conflict');
+      }
+      if (request.status === 'Rejected') {
+        return Response.errorResponse(res, 409, 'Request already rejected', 'conflict');
+      }
+      const requester = await UserService.findUserById(requesterId);
+      const data = await requestService.rejectUpdateRequest(requestId, 'Rejected');
+      const msg = email.rejectAcceptRequestTemplate(reason, manager, requester, 'Request rejected');
+      await email.sendmail(msg);
+      return Response.customResponse(res, 200, 'Request rejected successfully', data);
+    } catch (error) {
+      return Response.errorResponse(res, 500, 'Something went wrong', error);
+    }
+  }
+
+  /**
+   * @param {object} req request
+   * @param {object} res response
+   * @return {function} requests
+   */
+  async acceptRequest(req, res) {
+    try {
+      const { reason } = req.body;
+      const manager = req.user;
+      const { requestId } = req.params;
+      const request = await requestService.getRequest({ id: requestId });
+      if (!request) {
+        return Response.errorResponse(res, 404, 'request not found', 'Not found');
+      }
+      const requesterId = request.user;
+      if (request.status === 'Approved') {
+        return Response.errorResponse(res, 409, 'Request already accepted', 'conflict');
+      }
+      if (request.status === 'Rejected') {
+        return Response.errorResponse(res, 409, 'Request already rejected', 'conflict');
+      }
+      const requester = await UserService.findUserById(requesterId);
+      const data = await requestService.rejectUpdateRequest(requestId, 'Approved');
+      const msg = email.rejectAcceptRequestTemplate(reason, manager, requester, 'Request approved');
+      await email.sendmail(msg);
+      return Response.customResponse(res, 200, 'Request approved successfully', data);
```

Comment:
I think in the data section we can return the data as `request id`: `1`. Instead of just writing out the request id

Reply:
I'll look into it.